OneTimePasswordTests/KeychainTests.swift

```diff
@@ -0,0 +1,215 @@
+//
+//  KeychainTests.swift
+//  OneTimePassword
+//
+//  Copyright (c) 2013-2015 OneTimePassword authors
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in all
+//  copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+//
+
+import XCTest
+import OneTimePassword
+
+let testToken = Token(
+    name: "Name",
+    issuer: "Issuer",
+    generator: Generator(
+        factor: .Timer(period: 45),
+        secret: NSData(base32String: "AAAQEAYEAUDAOCAJBIFQYDIOB4"),
+        algorithm: .SHA256,
+        digits: 8
+    )!
+)
+
+class KeychainTests: XCTestCase {
+    let keychain = Keychain.sharedInstance
+
+    func testPersistentTokenWithIdentifier() {
```

Comment:
Length Violation: Function body should be span 40 lines or less: currently spans 57 lines